crates/apollo-parser/src/parser/mod.rs

```diff
@@ -146,12 +146,15 @@ impl<'a> Parser<'a> {
 
         match builder {
             syntax_tree::SyntaxTreeWrapper::Document(tree) => tree,
-            syntax_tree::SyntaxTreeWrapper::FieldSet(_) => {
+            syntax_tree::SyntaxTreeWrapper::Type(_)
+            | syntax_tree::SyntaxTreeWrapper::FieldSet(_) => {
                 unreachable!("parse constructor can only construct a document")
             }
         }
     }
 
+    /// Parse selection set tokens. Useful for specifically parsing selections
+    /// sets which are part of specific directives, like `@requires`.
```

Comment:
```suggestion
    /// Parse a selection set with optional outer braces.
    /// This is the expected format of the string value of the `fields` argument of some directives
    /// like [`@requires`](https://www.apollographql.com/docs/federation/federated-types/federated-directives/#requires).
```

Reply:
Braces? Brackets?

Reply:
IMO braces means `{}`, and brackets can mean that or one of the other pairs